cloud/deployment/workerqueue/workerqueue.go

```diff
@@ -0,0 +1,165 @@
+	}
+
+	if wQueueMax != 0 {
+		// use the value from the user input
+		workerQueueToCreate.MaxWorkerCount = wQueueMax
+	} else {
+		// set default value as user input did not have it
+		workerQueueToCreate.MaxWorkerCount = workerQueueDefaultOptions.MaxWorkerCount.Default
+	}
+
+	if wQueueConcurrency != 0 {
+		workerQueueToCreate.WorkerConcurrency = wQueueConcurrency
+	} else {
+		workerQueueToCreate.WorkerConcurrency = workerQueueDefaultOptions.WorkerConcurrency.Default
+	}
+	return workerQueueToCreate
+}
+
+// GetWorkerQueueDefaultOptions calls the workerqueues query
+// It returns WorkerQueueDefaultOptions if the query succeeds
+// An error is returned if it fails
+
+func GetWorkerQueueDefaultOptions(client astro.Client) (astro.WorkerQueueDefaultOptions, error) {
+	var (
+		workerQueueDefaultOptions astro.WorkerQueueDefaultOptions
+		err                       error
+	)
+	workerQueueDefaultOptions, err = client.GetWorkerQueueOptions()
+	if err != nil {
+		return astro.WorkerQueueDefaultOptions{}, err
+	}
+	return workerQueueDefaultOptions, nil
+}
+
+// IsWorkerQueueInputValid checks if the requestedWorkerQueue adheres to the floor and ceiling set in the defaultOptions
+// if it adheres to them, it returns nil
+// ErrInvalidWorkerQueueOption is returned if min, max or concurrency are out of range
+
+func IsWorkerQueueInputValid(requestedWorkerQueue *astro.WorkerQueue, defaultOptions astro.WorkerQueueDefaultOptions) error {
+	var errorMessage string
+	if !(requestedWorkerQueue.MinWorkerCount >= defaultOptions.MinWorkerCount.Floor) ||
+		!(requestedWorkerQueue.MinWorkerCount <= defaultOptions.MinWorkerCount.Ceiling) {
+		errorMessage = fmt.Sprintf("min worker count must be between %d and %d", defaultOptions.MinWorkerCount.Floor, defaultOptions.MinWorkerCount.Ceiling)
+		return errors.Wrap(ErrInvalidWorkerQueueOption, errorMessage)
+	}
+	if !(requestedWorkerQueue.MaxWorkerCount >= defaultOptions.MaxWorkerCount.Floor) ||
+		!(requestedWorkerQueue.MaxWorkerCount <= defaultOptions.MaxWorkerCount.Ceiling) {
+		errorMessage = fmt.Sprintf("max worker count must be between %d and %d", defaultOptions.MaxWorkerCount.Floor, defaultOptions.MaxWorkerCount.Ceiling)
+		return errors.Wrap(ErrInvalidWorkerQueueOption, errorMessage)
+	}
+	if !(requestedWorkerQueue.WorkerConcurrency >= defaultOptions.WorkerConcurrency.Floor) ||
+		!(requestedWorkerQueue.WorkerConcurrency <= defaultOptions.WorkerConcurrency.Ceiling) {
+		errorMessage = fmt.Sprintf("worker concurrency must be between %d and %d", defaultOptions.WorkerConcurrency.Floor, defaultOptions.WorkerConcurrency.Ceiling)
+		return errors.Wrap(ErrInvalidWorkerQueueOption, errorMessage)
+	}
+	return nil
+}
+
+// QueueExists takes a []existingQueues and a queueToCreate as arguments
+// It returns true if queueToCreate exists in []existingQueues
+// It returns false if queueToCreate does not exist in []existingQueues
+
+func QueueExists(existingQueues []astro.WorkerQueue, queueToCreate *astro.WorkerQueue) bool {
```

Comment:
Vaguely remember discussing handling the ID in the create / update paths, do we expect an ID in the create command? (Due to the  WQ struct being shared)

Reply:
we will handle updates in #670. For now `QueueExists()` is using `ID` and `Name` to see if the queue to create exists. If it does, it fails with an `ErrCannotUpdateExistingQueue`